src/composeRestyleFunctions.ts

```diff
@@ -46,15 +48,20 @@ const composeRestyleFunctions = <
       dimensions: Dimensions;
     },
   ): RNStyle => {
-    const styles = funcs.reduce((acc, func) => {
```

Comment:
Each style function in `funcs` takes all the props of the component, applies a transformation and returns a style attribute that then will be used to create a stylesheet. If there is no value to transform, each function returns an empty object instead.

Here, we were running every style function every time even though, not all transformations were needed. Example:

`<Box marginTop="m" />`

Here we only need to transform `marginTop="m"` to `marginTop: 16`, however we run all the transformations registered for the Box component here: https://github.com/sbalay/restyle-benchmark/blob/master/src/restyle/createBox.ts#L47

So we also check for `backgroundColor` and return an empty object, we also check for `marginBottom` and returns an empty object, we also check for `paddingHorizontal` and so on.

With the change below, we only run transformation functions for the props that are present. In the example above we would only run the function that transforms `marginTop="m"` to `{marginTop: 16}`